addon-irsa.tf

```diff
@@ -0,0 +1,29 @@
+# IMPORTANT: This file is synced with the "terraform-aws-eks-universal-addon" module. Any changes to this file might be overwritten upon the next release of that module.
+module "addon-irsa" {
+  for_each = local.addon_irsa
+
+  source = "git::https://github.com/lablabs/terraform-aws-eks-universal-addon.git//modules/addon-irsa?ref=v0.0.3"
+
+  enabled = var.enabled
+
+  cluster_identity_oidc_issuer     = var.cluster_identity_oidc_issuer != null ? var.cluster_identity_oidc_issuer : try(each.value.cluster_identity_oidc_issuer, "")
+  cluster_identity_oidc_issuer_arn = var.cluster_identity_oidc_issuer_arn != null ? var.cluster_identity_oidc_issuer_arn : try(each.value.cluster_identity_oidc_issuer_arn, "")
+
+  rbac_create               = var.rbac_create != null ? var.rbac_create : try(each.value.rbac_create, true)
+  service_account_create    = var.service_account_create != null ? var.service_account_create : try(each.value.service_account_create, true)
+  service_account_name      = var.service_account_name != null ? var.service_account_name : try(each.value.service_account_name, each.key)
+  service_account_namespace = var.service_account_namespace != null ? var.service_account_namespace : try(each.value.service_account_namespace, local.addon_namespace)
+
+  irsa_role_create      = var.irsa_role_create != null ? var.irsa_role_create : try(each.value.irsa_role_create, true)
+  irsa_role_name_prefix = var.irsa_role_name_prefix != null ? var.irsa_role_name_prefix : try(each.value.irsa_role_name_prefix, "${each.key}-irsa")
+  irsa_role_name        = var.irsa_role_name != null ? var.irsa_role_name : try(each.value.irsa_role_name, local.addon_helm_chart_name)
```

Comment:
What happens if someone does not use a unique `irsa_role_name` for each component? Won't it conflict with the fact that the role will already exist since `addon_helm_chart_name` is used by default?